h2o-core/src/main/java/water/rapids/ast/prims/string/algorithms/LevenstheinDistanceComparator.java

```diff
@@ -0,0 +1,79 @@
+package water.rapids.ast.prims.string.algorithms;
+
+import no.priv.garshol.duke.Comparator;
+
+import static org.apache.commons.lang.math.IEEE754rUtils.min;
+
+/**
+ * Non-weighted, case-sensitive Levensthein distance implementation inspired by Simmetrics library.
+ */
+public class LevenstheinDistanceComparator implements Comparator {
+
+  @Override
+  public boolean isTokenized() {
+    return true;
+  }
+
+  @Override
+  public double compare(String a, String b) {
+    if (a.isEmpty() && b.isEmpty()) {
+      return 1D;
+    }
+
+    return 1D - (distance(a, b) / Math.max(a.length(), b.length()));
+  }
+
+
+  /**
+   * Computes a case-sensitive Levensthein distance of two strings
+   *
+   * @param a First compared instance of {@link String}
+   * @param b Second compared instance of {@link String}
+   * @return Computed distance between two given strings
+   */
+  private double distance(final String a, final String b) {
```

Comment:
What is the performance of this implementation compared to the original implementation from Duke?

Reply:
Some hard numbers, as the Big-O is the same. It takes 41 seconds to compare 10 million times a 32-char long:

```java
levenshteinDistanceComparator.compare("AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
          "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa");
```

I takes 8 seconds to compare 3-char long string 100 /one hundred/ million times:
```java
      test = levenshteinDistanceComparator.compare("AAAA",
          "aaa");
```

To compare a 1 char long string, it takes half a second for 100 /one hundred/ million times of comparison.

No warmup done.